packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/AudienceView.tsx

```diff
@@ -25,12 +34,58 @@ export interface AudienceViewProps extends HasContainerId {
 export function AudienceView(props: AudienceViewProps): React.ReactElement {
 	const { containerId } = props;
 
-	// TODO: Post message requesting Audience summary
-	// TODO: Listen for Audience data updates
+	const messageRelay = React.useContext(MessageRelayContext);
+
+	if (messageRelay === undefined) {
+		throw new Error(
+			"MessageRelayContext was not defined. Parent component is responsible for ensuring this has been constructed.",
+		);
+	}
+
+	const [allAudienceMembers, setAllAudienceMembers] = React.useState<
+		Map<string, IClient> | undefined
+	>();
+	const [audienceHistory, setAudienceHistory] = React.useState<readonly unknown[]>();
 
 	React.useEffect(() => {
-		// TODO
-	}, [containerId]);
+		/**
+		 * Handlers for inbound messages related to Audience
+		 */
+
+		const inboundMessageHandlers: InboundHandlers = {
+			["AUDIENCE_EVENT"]: (untypedMessage) => {
+				const message: AudienceEventMessage = untypedMessage as AudienceEventMessage;
+				console.log("Passed 3");
+				console.log(message);
+
+				// setAllAudienceMembers([message.data.audienceState]);
+				setAudienceHistory([message.data.audienceHistory]);
+				return true;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the Message Relay
+		 */
+
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [
+		containerId,
+		allAudienceMembers,
```

Comment:
You'll want to remove `allAudienceMembers` and `audienceHistory` from this list. Otherwise we will rebind the message handlers each time we get a data update, which we don't want. We only want to change our listeners when the underlying info they depend on changes. I.e. 

- The containerId the view is currently associated with (so we can ensure we're only requesting data / listening to updates for the correct debugger instance)
- The 2 `set` functions, since we call them from within the hook.

Reply:
Changed the list to `[containerId, setClientId, setAudienceState, setAudienceHistory]`.